client/state/sites/hooks/use-site-options.ts

```diff
@@ -0,0 +1,29 @@
+import { snakeToCamelCase } from '@automattic/js-utils';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import getSiteOptions from '../selectors/get-site-options';
+
+type SiteOptions = { [ key: string ]: any };
+
+/**
+ * Get the site options
+ *
+ * @param optionNames options you want to query
+ * @returns {object} site options and the properties are listed in the given option names and converted to camel case
+ */
+const useSiteOptions = ( optionNames: string[] = [] ): SiteOptions => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const siteOptions: SiteOptions | null = useSelector( ( state ) =>
+		getSiteOptions( state, selectedSiteId || 0 )
+	);
+
+	const result: SiteOptions = {};
+
+	optionNames.forEach( ( optionName ) => {
+		result[ snakeToCamelCase( optionName ) ] = siteOptions?.[ optionName ] ?? null;
+	} );
+
+	return result;
```

Comment:
This hook is guaranteed to return a different object each time it's invoked, and I'm wondering whether it'd be possible to only have it return a fresh object when the value of the site options actually changes? Otherwise this hook could cause whatever component it's part of to re-render on every store change. Even when the store change wasn't relevant.

Alternatively maybe it could just be a `useSiteOption()` hook? That only returns the value of a single option? Analogous to [the `getSiteOption()` selector](https://github.com/Automattic/wp-calypso/blob/84c520f6447ad9cc6c81df70f5a78f55fb1a6d91/client/state/sites/selectors/get-site-option.js).

Reply:
Let me think how to avoid re-render on every store change 🤔